components/prism-ruby.js

```diff
@@ -102,6 +102,21 @@
 			inside: {
 				'interpolation': interpolation
 			}
+		},
+		{
+			pattern: /<<([a-z_]\w*)[\r\n](?:.*[\r\n])*?\1/i,
```

Comment:
Couldn't we easily add support for the other forms like this?

(The `delimiter` pattern has to be adjusted as well.)

```suggestion
			pattern: /<<[-~']?([a-z_]\w*)'?[\r\n](?:.*[\r\n])*?\1/i,
```

Reply:
I was wondering about that. One of the forms disables interpolation, though:

```
<<'STRING'
will #{not} be interpolated
STRING
```

Can you vary whether interpolation is enabled by the contents of the match?

Reply:
I was also hoping to add support for the heredoc anchor indicating the contents, like:

```
<<~SQL
  SELECT * FROM table WHERE column = 'value';
SQL
```

This is common in IDE syntax these days, and similar to what prism does for code fences in markdown which can contain other languages:

~~~
```sql
SELECT * FROM table WHERE column = 'value';
```
~~~

Reply:
> Can you vary whether interpolation is enabled by the contents of the match?

Yes and no. You can't change the `inside` tokens based on the match but you can just use two tokens like this:

```js
{
	pattern: /<<[-~`]?([a-z_]\w*)`?[\r\n](?:.*[\r\n])*?\1/i,
	alias: 'heredoc-string',
	greedy: true,
	inside: {
		'delimiter': {
			pattern: /^<<[a-z_]\w*|[a-z_]\w*$/i,
			alias: 'symbol',
			inside: {
				'punctuation': /^<<([-~]?)/
			}
		},
		'interpolation': interpolation
	}
},
{
	pattern: /<<'([a-z_]\w*)'[\r\n](?:.*[\r\n])*?\1/i,
	alias: 'heredoc-string',
	greedy: true,
	inside: {
		'delimiter': {
			pattern: /^<<[a-z_]\w*|[a-z_]\w*$/i,
			alias: 'symbol',
			inside: {
				'punctuation': /^<<([-~]?)/
			}
		}
	}
}
```

> I was also hoping to add support for the heredoc anchor indicating the contents

That's more difficult to do, especially when interpolation comes into play. 

Let's add support for heredoc first and do content highlighting later.